packages/bot-web-ui/src/stores/journal-store.js

```diff
@@ -105,6 +130,11 @@ export default class JournalStore {
 
     @action.bound
     clear() {
-        this.unfiltered_messages.clear();
+        this.unfiltered_messages = this.unfiltered_messages.slice(0, 0); // force array update
+    }
+
+    @action.bound
+    disposeJournalListeners() {
```

Comment:
Same here, perhaps we can define an `onUnmount` in `journal-store.js` that we'll call in `run-panel-store.js`, it would allow us to avoid patterns like this.